src/ui/viewmodels/MessageBoxViewModel.hh

```diff
@@ -0,0 +1,167 @@
+#ifndef RA_UI_MESSAGEBOX_VIEW_MODEL_H
+#define RA_UI_MESSAGEBOX_VIEW_MODEL_H
+#pragma once
+
+#include "ui/WindowViewModelBase.hh"
+
+namespace ra {
+namespace ui {
+namespace viewmodels {
+
+enum class MessageBoxButtons
+{
+    OK,
+    OKCancel,
+    YesNo,
+    YesNoCancel,
+    RetryCancel,
+};
+
+enum class MessageBoxIcon
+{
+    None,
+    Info,
+    Warning,
+    Error,
+};
+
+class MessageBoxViewModel : public WindowViewModelBase
+{
+public:
+    MessageBoxViewModel() noexcept {}
+
+    MessageBoxViewModel(const std::wstring& sMessage) noexcept
+    {
+        SetMessage(sMessage);
+    }
+
+#pragma push_macro("GetMessage") // windows stole my method name
```

Comment:
once we fully decouple the UI from the ViewModels, the conflict should no longer exist.